File: week01/1. Two Sum.cpp
// T - O(n^2), M - O(1)
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for(int i=0;i<nums.size()-1;i++){
            for(int j=i+1;j<nums.size();j++){
                int sum = nums[i]+nums[j];
                if(sum==target){
                    return {i,j};
                }
            }
        }
        return {};
    }
};
/**....................................................................**/

// T - O(n), M - O(n)
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> seen;
        for(int i=0;i<nums.size();i++){
            int remaining = target-nums[i];
            if(seen.count(remaining)){
                return {i,seen[remaining]};
            }
            seen[nums[i]]=i;
        }
        return {};
    }
};

File: week01/1.two_sum.py
# problem link: https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lists = {}
        
        for i in range(len(nums)):
            seen = target - nums[i]
            if seen in lists:
                return [lists[seen], i]
            lists.update({nums[i]: i})
            
        return []